system/lib/wasmfs/file.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+// This file defines the file object of the new file system.
+// Current Status: Work in Progress.
+// See https://github.com/emscripten-core/emscripten/issues/15041.
+
+#pragma once
+
+#include <emscripten/html5.h>
+#include <mutex>
+#include <wasi/api.h>
+
+class File {
+  // TODO: Add other File properties later.
+
+  // Mutex is used to lock File during multi-threaded access.
+  std::mutex mutex;
```

Comment:
These mutexes can be removed from `File` and `OpenFileDescriptor`.

Reply:
Yep, was waiting on integrating with `Lockable`. Will do that.